Задача: Потренируем StreamAPI 1.

piligrim-curious opened the pull request:
**Main**: Демонстрация работы;
**ListOperations**: Класс с функциями.

piligrim-curious committed:
Задача: Потренируем StreamAPI 1.
Main: Демонстрация работы;
ListOperations: Класс с функциями.

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+        return numbers.stream()
+                .mapToInt(Integer::intValue)
+                .max()
+                .orElse(Integer.MIN_VALUE);
```
Возвращение 0 может привести к путанице, так как это может быть воспринято как реальное максимальное значение. Лучше либо возвращать Optional, либо выбрасывать исключение для пустого списка.

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+    static double findAverage(List<Integer> numbers) {
+        return numbers.stream()
+                .mapToDouble(Integer::doubleValue)
+                .sum() / numbers.size();
```
можно воспользоваться методом average это более практичный вариант

```java
  return numbers.stream()
                .mapToInt(Integer::intValue)
                .average()
                .orElse(0.0);

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+        return strings.stream().allMatch(filter);
+    }
+
+    static int findMinGreaterThan(List<Integer> numbers, int condition) {
```
вместо condition переменую можно назвать threshold - более прозрачное название для порогового числа

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+        return numbers.stream()
+                .filter(number -> number > condition)
+                .min(Integer::compareTo)
+                .orElse(0);
```
аналогично max  - "Возвращение 0 может привести к путанице, так как это может быть воспринято как реальное максимальное значение. Лучше либо возвращать Optional, либо выбрасывать исключение для пустого списка."

MarselYsup commented on src/main/java/school/faang/training/streamapi1/Main.java:
```diff
@@ -0,0 +1,23 @@
+        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6);
+        List<String> strings = Arrays.asList("apple", "banana", "cherry", "date");
+
+        System.out.println("Сумма четных чисел: " + ListOperations.sumOfEvenNumbers(numbers));
```
тут и ниже некритично, но лучше использовать форматирование строк вместо конкатенации, так как это экономит память.

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+                .count();
+    }
+
+    static List<String> filterStringsContainingSubstring(List<String> strings, String ch) {
```
ch не определяет назначение аргумента, лучше назови substring

MarselYsup commented on src/main/java/school/faang/training/streamapi1/ListOperations.java:
```diff
@@ -0,0 +1,63 @@
+                .sum() / numbers.size();
+    }
+
+    static long countStringsStartingWith(List<String> strings, char ch) {
```
ch не определяет назначение аргумента, лучше назови symbol

piligrim-curious committed:
Задача: Потренируем StreamAPI 1.
Main: Демонстрация работы;
ListOperations: Класс с функциями.

Изменения:
(Main.java)
* Изменён вывод на f-строки;
* Добавлены исключения.
(ListOperations.java)
* Изменено название некоторых переменных;
* Некоторые функции претерпели изменения.

piligrim-curious committed:
Задача: Потренируем StreamAPI 1.
Main: Демонстрация работы;
ListOperations: Класс с функциями.

Изменения:
(Main.java)
* Изменён вывод на f-строки;
* Добавлены исключения.
(ListOperations.java)
* Изменено название некоторых переменных;
* Некоторые функции претерпели изменения.
